src/stream/src/executor/merge.rs

```diff
@@ -200,12 +217,28 @@ impl Executor for MergeExecutor {
     }
 }
 
+#[derive(Default)]
+struct StagedWatermarks {
+    in_heap: bool,
+    staged: VecDeque<Watermark>,
+}
+
+struct UntransferredWatermarks {
+    // We store the smallest watermark of each upstream, because the next watermark to emit is
+    // among them.
+    pub first_untransferred_watermarks: BinaryHeap<Reverse<(Watermark, ActorId)>>,
+    // We buffer other watermarks of each upstream. The next-to-smallest one will become the
+    // smallest and be moved into heap
+    pub other_untransferred_watermarks: HashMap<ActorId, StagedWatermarks>,
```

Comment:
```suggestion
    pub other_untransferred_watermarks: BTreeMap<ActorId, StagedWatermarks>,
```

Reply:
> Suggested change 
>     pub other_untransferred_watermarks: HashMap<ActorId, StagedWatermarks>,
>     pub other_untransferred_watermarks: BTreeMap<ActorId, StagedWatermarks>,

`remove_upstream` is rare so I think `HashMap` is fine

Reply:
Since the number of actors is small, indexing the btree-map will also be much faster.

Reply:
> Since the number of actors is small, indexing the btree-map will also be much faster than the hash map.

thanks. I learn more about rust. I thought `HashMap` is O(1) so it's efficient in the past. However, hash costs.

Reply:
> Since the number of actors is small, indexing the btree-map will also be much faster than the hash map.

fixed.